Northstar.Custom/mod/scripts/vscripts/sh_message_utils.gnut

```diff
@@ -0,0 +1,403 @@
+	RuiSetFloat2( rui, "offset", <0,-250,0> )
+	
+	statusMessageTable[ id ] <- rui
+}
+
+void function AnnouncementMessageHandler_Threaded()
+{
+	while( true )
+	{
+		while( announcementMessageList.len() == 0 )
+			WaitFrame()
+
+		AnnouncementFromClass( GetLocalViewPlayer(), announcementMessageList[0] )
+
+		wait 5
+
+		announcementMessageList.remove(0)
+	}
+}
+
+void function ShowAnnouncementMessage()
+{
+	AnnouncementData announcement = Announcement_Create( ServerMessage.title )
+	Announcement_SetSubText( announcement, ServerMessage.description )
+	Announcement_SetTitleColor( announcement, ServerMessage.color )
+	Announcement_SetPurge( announcement, true )
+	Announcement_SetPriority( announcement, ServerMessage.priority )
+	Announcement_SetSoundAlias( announcement, SFX_HUD_ANNOUNCE_QUICK )
+	Announcement_SetStyle( announcement, ServerMessage.style )
+	
+	announcementMessageList.append( announcement )
+}
+
+void function PopUpMessageHandler_Threaded()
+{
+	while( true )
+	{
+		while( popupMessageList.len() == 0 )
+			WaitFrame()
+
+		var rui = CreateCockpitRui( $"ui/killdeath_info.rpak" )
+		RuiSetGameTime( rui, "startTime", Time() )
+		RuiSetFloat( rui, "duration", 20 ) // It has a weird end animation
+		RuiSetString( rui, "messageText", popupMessageList[0].description )
+		RuiSetBool( rui, "isBigText", true )
+
+		wait 2.4
+
+		popupMessageList.remove(0)
+		RuiDestroyIfAlive( rui )
+	}
+}
+
+void function ShowPopUpMessage()
+{
+	popupMessageList.append( clone ServerMessage )
+}
+
+void function LargeMessageHandler_Threaded()
+{
+	while( true )
+	{
+		while( largeMessageList.len() == 0 )
+			WaitFrame()
+
+		var rui = CreatePermanentCockpitRui( $"ui/fd_tutorial_tip.rpak" )
+		RuiSetImage( rui, "backgroundImage", NSAssetFromString( largeMessageList[0].image ) )
+		RuiSetString( rui, "titleText", largeMessageList[0].title )
+		RuiSetString( rui, "descriptionText", largeMessageList[0].description )
+		RuiSetGameTime( rui, "updateTime", Time() )
+		RuiSetFloat( rui, "duration", largeMessageList[0].duration )
+
+		wait largeMessageList[0].duration * 0.34
+
+		largeMessageList.remove(0)
+		RuiDestroyIfAlive( rui )
+	}
+}
+
+void function ShowLargeMessage()
+{
+	largeMessageList.append( clone ServerMessage )
+}
+
+void function InfoMessageHandler_Threaded()
+{
+	while( true )
+	{
+		while( infoMessageList.len() == 0 )
+			WaitFrame()
+
+		var rui = CreatePermanentCockpitRui( $"ui/death_hint_mp.rpak" )
+		RuiSetString( rui, "hintText", infoMessageList[0].description )
+		RuiSetGameTime( rui, "startTime", Time() )
+		RuiSetFloat3( rui, "bgColor", <0,0,0> )
+		RuiSetFloat( rui, "bgAlpha", 0.5 )
+
+		wait 7
+
+		infoMessageList.remove(0)
+		RuiDestroyIfAlive( rui )
+	}
+}
+
+void function ShowInfoMessage()
+{
+	infoMessageList.append( clone ServerMessage )
+}
+
+#endif // CLIENT
+
+asset function NSAssetFromString( string str )
+{
+	return expect asset(compilestring(format("return $\"%s\"", str))())
+}
```

Comment:
since now StringToAsset is merged into Launcher this is not required anymore

Reply:
@F1F7Y I'm on vacation, so could you do this? :P

Reply:
Currently coming back from vacation, will push this when I have time

Reply:
done